# CSE110 Assignment 2 

This assignment is to guide in solving basic programming problems using loop in Python. 




##Write your name, student ID and theory section number below:


In [ ]:
#STUDENT NAME: Murshed Jamil Alif
#STUDENT ID: 21301069
#CSE110 SECTION: 07 

## Write the code in Python for following problems:

**<font color='red'>[Please follow variable naming conventions including using meaningful variable names for all the tasks]</font>**

### Task 1

Write python program, which prints the following sequences of values in loops:

a)	24, 18, 12, 6, 0, -6\
b)	-10, -5, 0, 5, 10, 15, 20\
c)	18, 27, 36, 45, 54, 63\
d)	18, -27, 36, -45, 54, -63

=====================================================================

**Hints(1):** Use a while loop for solving these problems.

**Hints(2):** We are already familiar with the print() function. But when we use it to print any value it automatically adds an additional newline after each print statement.

For example:\
print(1)\
print(2)

Output:\
1\
2

=====================================================================

To solve this problem, in Python3, we can add an extra argument (end = " ") in the print function which tells the program to skip printing the additional newline.

For example:\
print(1, end =" ")\
print(2)

Output: (prints the next output right to the previous one)\
12

=====================================================================

In Task-1(a), the loop counter should be initialized at 24 and the loop should terminate when the loop counter reaches -6. The difference between the first two values is 24 - 18 = 6. So the loop counter value is getting decremented by 6 in every iteration.

For your understanding, task 1(a) code is done for you.


In [30]:
# a) 24, 18, 12, 6, 0, -6

# initialize loop counter
counter = 24

# loop structure
while counter >= -6:
    
    #inside loop body
    if counter == -6:
        print(counter, end = "") 
    else:
        print(counter, end = ", ") 
     
    counter = counter - 6 #updating loop counter
    #inside loop body
    
#outside loop body

24, 18, 12, 6, 0, -6

In [31]:
counter =-10
while counter <= 20:

    if counter == 20:
     print(counter, end="")
    else:
     print(counter, end=", ")
    counter = counter + 5

-10, -5, 0, 5, 10, 15, 20

In [32]:
# c)18, 27, 36, 45, 54, 63

counter =18
while counter <= 63:

    if counter == 63:
     print(counter, end="")
    else:
     print(counter, end=", ")
    counter = counter + 9

18, 27, 36, 45, 54, 63

**Hints for 1(d):** 

print(5 * (-1)) gives output -5\
print("-" + str(5)) gives output -5

In [34]:
# d)18,-27,36,-45,54,-63
counter =18
while counter <= 63:
    if(counter==63):
        print(counter*-1,end="")
        break

    elif(counter%2)==0:
        print(counter,end=",")
    
    else:
        print(counter*-1,end=",")
    counter+=9    
        


18,-27,36,-45,54,-63

### Task 2
Write a Python code for the following:

1)	Ask the user to enter the name of his favorite car.\
2)	Ask the user to enter a Number

Display the name of the user’s favorite car, the number of times specified in the second step.

==========================================================

**Example 01:** If the user enters “Toyota” and 2, your program should print the name “Toyota” two times.

**Input:**\
Toyota\
2

**Output:**\
Toyota\
Toyota

==========================================================

**Example 02:** If the user enters “Veyron” and 5, your program should print the name “Veyron” five times.

**Input:**\
Veyron\
5

**Output:**\
Veyron\
Veyron\
Veyron\
Veyron\
Veyron


In [36]:
#Todo
car_name=input("")
num=int(input())

for i in range(num):
  print(car_name)


Kawasaki Ninja H2R
10
Kawasaki Ninja H2R
Kawasaki Ninja H2R
Kawasaki Ninja H2R
Kawasaki Ninja H2R
Kawasaki Ninja H2R
Kawasaki Ninja H2R
Kawasaki Ninja H2R
Kawasaki Ninja H2R
Kawasaki Ninja H2R
Kawasaki Ninja H2R


### Task 3
Write a Python code of a program that adds all numbers that are multiples of **both 7 and 9** up to 600 (including 600).

Output: 2835


In [39]:
#Todo
i=1
sum=0
while i<=600:
 if(i%7==0 and i%9==0):
      sum=sum+i
      i=i+1
 else:
  i = i + 1
print(sum)
       
        

2835


### Task 4
Write a Python code of a program that adds all numbers that are multiples of **either 7 or 9** up to 600(including 600). \
Ensure that numbers like 63 are added only once in the sum.

Output: 42649


In [16]:
#Todo
sum=0
for i in range(601):
 if i%7==0 or i%9==0:
    sum=sum+i
 else:
    i=i+1
print(sum)    
 

42649


### Task 5
Write a Python code of a program that adds all numbers that are multiples of **either 7 or 9 but not both**, up to 600(including 600).

Output: 39814


In [1]:
#Todo
sum=0
for i in range(601):
 if i%7==0 and i%9!=0:
        sum=sum+i
       
 elif i%7!=0 and i%9==0:
    sum=sum+i
print(sum)    
    

39814


### Task 6
Write a Python code to displays all the **odd numbers** between 10 and 50.

Output: 11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49


In [40]:
#Todo
for i in range(11,50,2):
 print(i,end=" ")


11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49 

### Task 7
Write a Python code that will calculate the **value of y if the expression** of y is as follows (n is the input):  

$y = 1^2 - 2^2 +3^2 -4^2 +5^2 .........+ n^2 $


==========================================================

**Example 01:** 

**Input:**\
10

**Output:**\
-55

==========================================================

**Example 02:** 

**Input:**\
20

**Output:**\
-210



In [48]:
#Todo
n=int(input("n:"))
sum=0
for i in range(n+1):
 if i%2!=0:
        sum=sum+(i**2)
 elif i%2==0:
    sum=sum+(-1)*(i**2)
print(sum)   


n:10
-55


### Task 8

Write a Python code of a program that asks the user to enter ten numbers then display the total and the average of **ONLY** the **odd numbers** among those ten numbers. <font color='red'>[Please do not use list for this task]</font>

==========================================================

**Example 01:** 

**Input:**\
1

2

3

4

5

6

7

8

9

10

**Output:**
Total is 25 and Average is 5.0


**Explanation:**

Total is 25 = (1+3+5+7+9) and Average is 25/5 = 5.0

==========================================================


**Example 02:** 

**Input:**\
-20

3

-5

40

-17

9

20

-8

99
  
-200

**Output:**

Total is 89 and Average is 17.8


**Explanation:**

Total is 89 = (3+(-5)+(-17)+9+99) and Average is 89/5 = 17.8



In [66]:
#Todo
sum=0
count=0
i=1
while(i<11):
 n=int(input())
 if n%2!=0:
  sum=sum+n
  i=i+1
  count=count+1  
 else:
  i=i+1
    
    
avg=sum/count
 

print("Total is",sum,end=" ")
print("and Average is",avg)
    



-20
3
-5
40
-17
9
20
-8
99
-200
Total is 89 and Average is 17.8


### Task 9
Write a Python code for the following:
* Ask the user to enter a Number, N
* Display the summation of multiples of 7 up to that number (**from 1 to N inclusive**)

==========================================================

**Example01:** 

**Input:**\
50

**Output:**\
196

**Explanation:**\
7 + 14 + 21 + 28 + 35 + 42 + 49 = 196

==========================================================

**Example02:** 

**Input:**\
75

**Output:**\
385

**Explanation:**\
7 + 14 + 21 + 28 + 35 + 42 + 49 + 56 + 63 + 70 = 385





In [10]:
# Todo
n = int(input())
sum = 0
i = 1
while (i * 7 <= n):
    sum = sum + (i * 7)
    i = i + 1
print(sum)
       
        
   


75
385


### Task 10
Write a Python code that will read 10 numbers from the user, and then print the first number, the sum of the first 2 numbers, the sum of the first 3 numbers, and so on up to the sum of 10 numbers.

==========================================================


For example,

The user enters 10, output 10, then \
The user enters 2, (10+2) = 12, output 12, then \
The user enters 4, (10+2+4) = 16, output 16, then \
The user enters 20, (10+2+4+20) = 36, output 36, and continues till the 10$^{th}$ input


In [ ]:
#Todo
sum=0
i=1
while(i<=10):
 n=int(input())
 sum=sum+n
 print(sum)

10
10
2
12
3
15
3
18
3
21
3
24
3
27
3
30
3
33
3
36





### Task 11
Write a Python program which takes a number and prints the digits from the unit place, then the tenth, then hundredth, etc. <font color='red'>(Right to Left)</font>

<font color='red'>[Consider the input number to be an INTEGER. You are not allowed to use String indexing for solving this task]</font>

**Example:** If the user gives 32768, then print 8, 6, 7, 2, 3 

=========================================================================

**Hint (1):**  The input() function, converts the input data to String data type by default. Therefore, please type cast it to an integer before proceeding further.


**Hint  (2):** First to get the digit from the right side, we can take the remainder of the number using modulus (%) operator i.e. mod 10 to get the rightmost digit and print it. For dropping the last digit, we can perform floor division by 10 on the number and then continue the same to print the other digits as shown below.

32768 % 10 = 8 

32768 // 10 = 3276 

Then,

3276 % 10 = 6

3276 // 10 = 327 

and so on

327 % 10 = 7 

327 // 10 = 32 

32 % 10 = 2 

32 // 10 = 3 

3 % 10 = 3 

3 // 10 = 0 

Done! When the number becomes 0 you can end your loop.



In [99]:
#Todo
n=int(input())
while(n>0):
    if n//10==0:
       print(n%10,end="") 
    else:
       print(n%10,end=",") 
      
    n=n//10

 

7896
6,9,8,7

### Task 12

Write a Python program that takes a number and prints how many digits are in that number. \
<font color='red'>[Consider the input number to be an INTEGER.]</font>

<font color='red'>[You are not allowed to use len() function]</font>

**Example:** If the user gives 9876, your program should print 4. 



**Hint:** We may keep floor dividing by ten and count how many times this can be done until the number becomes 0 as illustrated below. 

9876 // 10 = 987, count = 1 \
then, 987 // 10 = 98, count = 2 \
98 // 10 = 9, count = 3 \
9 // 10 = 0, count = 4 

Done! When the number becomes 0 your loop can end giving you the count of digits, in this case for our input of 9876, 4 digits.



In [37]:
#Todo
n=int(input())
cout=0
while(n>0):
 reminder=n%10
 cout+=1
    
 a=n//10
 n=a
 
print(cout)

7896
4


### Task 13
Write a Python program that takes a number from the user and prints its digits from  <font color='red'> left to right. </font>

<font color='red'>[Consider the input number to be an INTEGER. You are not allowed to use String indexing for solving this task]</font>

=========================================================================

**Example:** if the user gives 32768, then print 3, 2, 7, 6, 8 

=========================================================================

**Hint(1):**  The input() function takes the input data as String data type by default. Please convert it to an integer before starting your code for the task.

**Hint(2):**  

Step 1: First, count how many digits\
Step 2: Then, calculate 10 to the power (number of digits - 1).

Step 3 with explanation: Say, the input given by the user as in our example, 32768 has 5 digits, so calculating 10 to the power 4 gives us 10000. Then floor dividing 32768 by 10000, we can get 3.\
Proceeding further, the remainder of 32768 by 10000 (32768 % 10000) gives us 2768. This time to get 2 we need to floor divide 2768 by 1000 which is basically our 10000/10. Again, taking remainder of 2768 by 1000 gives us 768 which we then divide by 100 (i.e. 1000/10) and keep on doing this until there are no more digits left (zero).


**To summarize and clarify:**\
Loop 1: First, we count digits like our Task 12, say 5 in this case for 32768\
Loop 2: Then, we calculate 10 to the power 4 (5-1), that is 10000.\
Loop 3: Then we keep repeating the three steps of floor dividing, modulus and dividing by 10 as demonstrated below.

32768 // 10000 = 3\
32768 % 10000 = 2768 \
10000 // 10 = 1000

2768 // 1000 = 2\
2768 % 1000 = 768\
1000 // 10 = 100

768 // 100 = 7\
768 % 100 = 68 \
100 // 10 = 10

68 // 10 = 6\
68 % 10 = 8 \
10 // 10 = 1

8 // 1 = 8\
8 % 1 = 0 \
1 // 10 = 0

Done. Loop ends as number has become 0.

In [1]:
n=int(input())
x=n
cout=0
while(x>0):
    x%10
    cout+=1
    x=x//10

while(cout>0):
    s=n//(10**(cout-1))
    
    if cout==1: 
     print(s,end="")
    else:
     print(s,end=",")  
    n=n%(10**(cout-1))
    cout-=1
    
    
    
    
    
    
    
    
    
 

    

37896
3,7,8,9,6

### Task 14

Write a Python program that takes a number as input from the user and prints the divisors of that number as well as how many divisors the number has.
<font color='red'>[The input number has to be an INTEGER]</font>

==========================================================

**Example 01:** 

**Input:**\
6

**Output:**\
1, 2, 3, 6 \
Total 4 divisors.


==========================================================

**Example 02:** 

**Input:**\
121

**Output:**\
1, 11, 121 \
Total 3 divisors.




In [98]:
#Todo
n=int(input())

total=0
i=1
while(i<=n):
 if n%i==0:
        total+=1
        if i==n:
         print(i,end="")
        else:
         print(i,end=",")   
        i+=1

 else:
    i+=1
    
print("\nTotal",total,"divisors" )

121
1,11,121
Total 3 divisors


### Task 15
Write a Python program that takes a number and tells if it is a perfect number or not. <font color='red'>[The input number has to be an INTEGER]</font>

**Perfect Number:** An integer number is said to be a perfect number if its factors, including 1 but not the number itself, sum to the number.

==========================================================

**Example 01:** 

**Input:**\
6

**Output:**\
6 is a perfect number

**Explanation:**\
6 has 4 divisors: 1, 2, 3, and 6. \
If we add all factors except 6 itself, 1 + 2 + 3 = 6. The sum of the factors excluding the number itself sum up to the number therefore "6 is a perfect number" is printed.

==========================================================


**Example 02:** 

**Input:**\
28

**Output:**\
28 is a perfect number

**Explanation:**\
28 has divisors: 1, 2, 4, 7, 14, and 28. \
If we add all factors except itself we get 28, 1 + 2 + 4 + 7 + 14 = 28.

==========================================================

**Example0 3:** 

**Input:**\
33

**Output:**\
33 is not a perfect number

**Explanation:**\
33 has 3 divisors: 1, 3, 11, and 33. \
If we add all factors except itself, 15 = 1 + 3 + 11. Sum not equal to the number, therefore 33 is not a perfect number.



In [104]:
#Todo
n=int(input())
sum=0
i=1
while(i<=n):
    if n%i==0:
        sum+=i
        i+=1
    else:
        i+=1
    
if (sum-n)==n:
    print(n, "is a perfect number")
else:
    print(n, "is not a perfect number")

28
28 is a perfect number


### Task 16

Write a Python program that asks the user for one number and tells if it is a prime number or not. <font color='red'>[The input number has to be an INTEGER]</font>

**Prime Number:** If a number has only two divisors, (1 and itself), then it is prime. If it is divisible by more numbers, then it is not a prime.

==========================================================

**Hint:** You may use the code for counting divisors from Task 14. 

==========================================================

**Example 01:** 

**Input:**\
11

**Output:**\
11 is a prime number

**Explanation:**\
11 has only 2 divisors: 1 and 11.

==========================================================

**Example 02:** 

**Input:**\
6

**Output:**\
6 is not a prime number

**Explanation:**\
6 has 4 divisors: 1, 2, 3 and 6.






In [2]:
#Todo
n=int(input())
i=2
while(i<=n):
    if (n==2):
        print(n,"is a prime number")
        break
        
    elif n%i!=0:
        print(n,"is a prime number")
        break 
    else:
        print(n,"is not a prime number")
        break
      

2
2 is a prime number


### Task 17

Write a Python program that asks the user for a quantity, then takes that many numbers as input and prints the maximum, minimum and average of those numbers. 

<font color='red'>[Please note that you CANNOT use max, min built-in functions]

[Also, you DO NOT need to use lists for this task]</font>

==========================================================

**Example 01:** If the user enters 5 as an input for quantity and the enters the 5 numbers, 10, 4, -1, -100, and 1. \
The output of your program should be: “Maximum 10”, “Minimum -100”, “Average is -17.2” as shown below.

**Input:**\
5
10
4
-1
-100
1

**Output:**\
Maximum 10\
Minimum -100\
Average is -17.2

**Explanation:**\
Average calculation: (10 + 4 + (-1) + (-100) + 1)/5 = -86/5 = -17.2



In [4]:
#Todo
n=int(input())
mx=0
mn=0
sum=0
i=0
while(i<n):
 num=int(input())
 sum+=num   
 if i==0:
        mx=num
        mn=num
        i+=1
       
 else:
    if(num>mx):
        mx=num
       
    elif(num<mn):
        mn=num
    i+=1
        
print("Maximum " ,mx)
print("Minimum " ,mn)
print("Average:", sum/n)        
        
    
    

5
10
4
-1
-100
1
Maximum  10
Minimum  -100
Average: -17.2


## Task 18

Write a python program that prints a square of size N using + where N will be given as input as illustrated in the examples below.

==========================================================

**Example 01:**

**Sample Input**

5

**Sample Output**<br>
+++++<br>
+++++<br>
+++++<br>
+++++<br>
+++++<br>


==========================================================

**Example 02:**

**Sample Input**

3

**Sample Output**<br>
+++<br>
+++<br>
+++<br>

In [8]:
#Todo
n=int(input())
for i in range(1,n+1):  #row handle kore
 for i in range(1,n+1): #column handle kore
    print("+",end="")
 print()   


3
+++
+++
+++


### Task 19

Write a python program that prints a rectangle of size M (height/line numbers) and N(length/column numbers) using incrementing numbers where M and N will be given as input. Please look it the examples below for clarification.

==========================================================

**Example 01:**

**Sample Input:**\
4\
6

**Sample Output:**<br>
123456<br>
123456<br>
123456<br>
123456<br>

==========================================================

**Example 02:**

**Sample Input:**\
3\
2

**Sample Output:**<br>
12<br>
12<br>
12<br>


In [15]:
#Todo
row=int(input())
column=int(input())

for i in range(1,row+1):
    for j in range(1,column+1):
        print(j,end="")
    print()

5
5
12345
12345
12345
12345
12345


### Task 20

Write a python program that prints a right angled triangle of height N using incrementing numbers where N will be given as input.

==========================================================

**Example 01:**

**Sample Input:**\
4

**Sample Output:**<br>
1<br>
12<br>
123<br>
1234<br>

==========================================================

**Example 02:**

**Sample Input:**\
5

**Sample Output:**<br>
1<br>
12<br>
123<br>
1234<br>
12345<br>

In [25]:
#Todo
row=int(input())
for i in range(1,row+1):  #row control loop
    for j in range(1,i+1): #column control loop
        print(j,end="")
    print()    
    

5
1
12
123
1234
12345


## Optional Tasks (21 -26) [Ungraded]

**These tasks are just for practice. No marks will be deducted for not completing them and no extra marks will be given for solving them. Just try and practice these advanced problems.**

### Task 21

Write a python program that prints all the fibonacci numbers from 0 to N where N will be given as input by the user. 

**A Fibonacci number is a number which is the summation of its previous two fibonacci numbers.**

First two fibonacci numbers are 0 and 1. 

So the 3rd is 0 + 1 = 1, 

the 4th is 1 + 1 = 2, 

the 5th is 1 + 2 = 3,

the 6th one is 2 + 3 = 5,

and so on. 

i.e. 0, 1, 1, 2, 3, 5, 8, 13, 21.......

==========================================================

**Example 01:**

**Sample Input:**<br>
10

**Sample Output:**<br>
0 1 1 2 3 5 8

==========================================================

**Example 02:**

**Sample Input:**<br>
15

**Sample Output:**<br>
0 1 1 2 3 5 8 13

In [29]:

nterms = int(input("Terms?:- "))


n1, n2 = 0, 1
count = 0


print("Fibonacci sequence:")
while count <= nterms:
       
        print(n1,end=" ")
        nth = n1 + n2
        # update values
        n1 = n2
        n2 = nth
        count=n1
  

Terms?:- 12
Fibonacci sequence:
0 1 1 2 3 5 8 

### Task 22

Write a python program that converts a Decimal Integer number to a Binary Number. 

**A decimal can be converted to a binary number by keeping track of the remainders after each division of that number by 2. **

==========================================================

**Example 01:**

For example, to convert 11 to a binary number, we can try the following approach.

11/2 = 5 (Remainder 1) 
<br>5/2 = 2 (Remainder 1) 
<br>2/2 = 1 (Remainder 0) 
<br>1/2 = 0 (Remainder 1) 

Now, taking the remainders from bottom to top gives us the binary number which is 1011. Binary of 11 is 1011.

**Sample Input:** <br>
11

**Sample Output:**<br>
1011

==========================================================

**Example 02:**

Similarly, to convert 13 to a binary number:

13/2 = 6 (Remainder 1) 
<br>6/2 = 3 (Remainder 0) 
<br>3/2 = 1 (Remainder 1) 
<br>1/2 = 0 (Remainder 1) 

Again, taking the remainders from the bottom to top, which is 1101 gives us the binary number. Binary of 13 is 1101.

**Sample Input:** <br>
13

**Sample Output:**<br>
1101

In [49]:
num = int(input("Decimal Value:-"))

sum = 0
i= 1
while num>0:
    rem = num%2
    sum = sum+(rem*i)
    i= i*10
    num = int(num/2)

print(" Binary Value: ", sum)

Decimal Value:-13
 Binary Value:  1101


### Task 23

Write a python program that converts a Binary number to a Decimal Integer Number after taking the binary number as an input from the user. 

**A binary number can be converted to its corresponding decimal number by multiplying each binary digit with a power of 2 where the power denotes the position of the binary digit**

==========================================================

**Example 01:**

For example, to convert 1010 to decimal, we can do this


$2^{3}$ x 1 + $2^{2}$ x 0 + $2^{1}$ x 1 + $2^{0}$ x 0 = 10


**Sample Input:** <br>
1010

**Sample Output:** <br>
10

==========================================================

**Example 02:**

So, to convert 11101, we can do this


$2^{4}$ x 1 + $2^{3}$ x 1 + $2^{2}$ x 1 + $2^{1}$ x 0 + $2^{0}$ x 1 = 29


**Sample Input:** <br>
11101

**Sample Output:** <br>
29

In [40]:
#Todo
num = list(input())
value= 0

for i in range(len(num)):
	digit =num.pop()
	if (digit=='1'):
		value = value + (i**2)
print("Decimal value:", value)


1010
Decimal value: 10


### Task 24
Write a Python program that asks the user for a range (a starting number and an ending number) and then 
**counts** how many numbers are prime numbers and how many numbers are perfect numbers between that range. It also **prints** those numbers in the format shown in the output of the example given below. 


=========================================================================

**Hint (1):** We may declare two strings/lists to store the prime and perfect numbers. Inside the loop, we check for prime and perfect numbers and add them to respective string/list besides counting them.

**Hint (2):** We may need to convert numbers in String to integers for checking and calculations. We may again need to convert numbers to Strings for storing and printing as the outputs shown in the example below.

=========================================================================

**Example:** If the user gives us a range of between 2 and 6, there are 3 prime numbers (2, 3, 5) and 1 perfect number (6) in that range. 

**Input:**\
2\
6

**Output:**\
Between 2 and 6,\
Found 3 prime numbers \
Found 1 perfect number\
Prime numbers: 2,  3,  5 \
Perfect number: 6 



In [38]:
n1=int(input())
n2=int(input())
s=""
m=""
for i in range(n1,n1+1):
    for y in i:
        if (i==2):
            

2
6


TypeError: 'int' object is not iterable